python/ray/data/_internal/execution/interfaces/execution_options.py

```diff
@@ -18,13 +18,15 @@ def __init__(
         cpu: Optional[float] = None,
         gpu: Optional[float] = None,
         object_store_memory: Optional[float] = None,
+        memory: Optional[float] = None,
         default_to_inf: bool = False,
     ):
         """Initializes ExecutionResources.
         Args:
             cpu: Amount of logical CPU slots.
             gpu: Amount of logical GPU slots.
             object_store_memory: Amount of object store memory.
+            memory: Amount of logical memory in bytes.
```

Comment:
Memory parameter used in logical resources is total memory (ie heap + object store) 

If we plan on keeping the object_store_memory separate, i think we'd rather make them non-overlapping

Reply:
Yeah, I agree it's not ideal that they overlap, but I don't think it's a major issue since this is consistent with the Ray Core semantics.

> we'd rather make them non-overlapping

How would I do that?

Reply:
Ray Core should exclude object store memory from the node memory logical resource.